packages/image/src/extractColors.ts

```diff
@@ -0,0 +1,78 @@
+interface PaletteOptions {
+  colorCount?: number;
+  quality?: number;
+}
+
+/**
+ * Extracts a palette of colors from a given image URL based on provided options.
+ * @param {string} imageUrl - The URL of the image.
+ * @param {PaletteOptions} [options] - The extraction options.
+ */
+export const extractColors = (imageUrl: string, options?: PaletteOptions) => {
+  return loadImageFromUrl(imageUrl)
+    .then(extractImage)
+    .then((data) => {
+      if (!data) {
+        throw new Error(
+          "Failed to extract pixel data from the image. Ensure the image URL is valid and accessible.",
+        );
+      }
+
+      const pixels = filterRelevantPixels(data, options?.quality);
+      const count = options?.colorCount || pixels.length;
+      return pixels.slice(0, count);
+    });
+};
```

Comment:
```suggestion
export const extractColors = (imageUrl: string, options?: PaletteOptions) =>
  loadImageFromUrl(imageUrl)
    .then(extractImage)
    .then((data) => {
      if (!data) {
        throw new Error(
          "Failed to extract pixel data from the image. Ensure the image URL is valid and accessible.",
        );
      }

      const pixels = filterRelevantPixels(data, options?.quality);
      const count = options?.colorCount || pixels.length;
      return pixels.slice(0, count);
    });
```

Reply:
I wanted to suggest this every time, so I made PR to force this automatically. How about this rule? https://github.com/fepack/image/pull/41

Reply:
[d29136d](https://github.com/fepack/image/pull/33/commits/d29136d8b0917088eabebfff82705910cd79f671) 😁